### PR TITLE
allow custom image normalization in CropModel

### LINKED ISSUE #1347: CropModel enforces ImageNet norm weights
As part of #1313 we are expanding how users can add CropModels, there is no reason that the weights have to follow ImageNet normalization. 

### Problem

`bounding_box_transform()` in `src/deepforest/datasets/cropmodel.py` hardcodes ImageNet normalization, making it impossible for users to supply models pretrained with different normalization stats.

```python
# Line 32 - no way to override this
data_transforms.append(resnet_normalize)
```

A user passing a custom model via `CropModel(model=my_model)` that expects different preprocessing will get silently wrong results.

### Proposed Fix

Add an optional `normalize` parameter to `bounding_box_transform` and thread it through from the CropModel config:

```python
def bounding_box_transform(augmentations=None, resize=None, normalize=None):
    if resize is None:
        resize = [224, 224]
    if normalize is None:
        normalize = resnet_normalize

    data_transforms = []
    data_transforms.append(transforms.ToTensor())
    data_transforms.append(normalize)
    data_transforms.append(transforms.Resize(resize))
    if augmentations:
        data_transforms.append(transforms.RandomHorizontalFlip(0.5))
    return transforms.Compose(data_transforms)
```

Users could then pass custom normalization via config:

```yaml
cropmodel:
    normalize:
        mean: [0.5, 0.5, 0.5]
        std: [0.5, 0.5, 0.5]
```

Or `normalize: false` to disable it entirely.

### PR DESCRIPTION
**Pull request by musaqlain**
## Desc
- `bounding_box_transform()` and `CropModel.normalize()` both hardcode ImageNet normalization (`mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]`) with no way for the user to override.
- A user passing a custom model via `CropModel(model=my_model)` that expects different preprocessing gets silently wrong results.
- Followed the threading pattern from #1313 (`resize_interpolation`) — adding an optional config parameter and passing it through both the training and prediction pipelines.

### Solution

The `normalize` config parameter supports three modes:

```yaml
# Default (null): ImageNet normalization
normalize:

# Disabled: skip normalization
normalize: false

# Custom: user-provided values
normalize:
    mean: [0.5, 0.5, 0.5]
    std: [0.5, 0.5, 0.5]
```

**1. Config.** Added `normalize` field (defaults to `null`) in [config.yaml](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/src/deepforest/conf/config.yaml#L120-L126) and [schema.py](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/src/deepforest/conf/schema.py#L110). The field is typed as `Any` since OmegaConf does not support unions of containers (`dict | bool`); validation happens in code.

**2. Training path.** [CropModel.normalize()](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/src/deepforest/model.py#L312-L343) now reads from config instead of returning hardcoded ImageNet values. [CropModel.get_transform()](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/src/deepforest/model.py#L198-L226) handles the case where `normalize()` returns `None` (disabled).

**3. Prediction path.** [bounding_box_transform()](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/src/deepforest/datasets/cropmodel.py#L16-L40) accepts an optional `normalize` parameter — `None` defaults to `resnet_normalize`, `False` skips it, or a `transforms.Normalize` instance is used directly. Threaded through [BoundingBoxDataset](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/src/deepforest/datasets/cropmodel.py#L60-L74) and [_predict_crop_model_()](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/src/deepforest/predict.py#L285-L292).

**4. Validation.** `CropModel.normalize()` raises a `ValueError` if only one of `mean` or `std` is provided, preventing silent misconfiguration.

**5. Tests.** 5 new tests added in [test_crop_model.py](https://github.com/musaqlain/DeepForest/blob/allow-custom-norm-weights/tests/test_crop_model.py#L210-L295):
- `test_crop_model_default_normalization` — verifies ImageNet stats when no config is set
- `test_crop_model_custom_normalization` — verifies custom `mean`/`std` are applied
- `test_crop_model_disabled_normalization` — verifies `normalize: false` skips normalization
- `test_crop_model_normalize_missing_mean_or_std` — verifies `ValueError` on incomplete config
- `test_bounding_box_transform_normalize` — verifies the prediction-path transform handles all three modes

**Breaking changes:** None. Default is `null` which applies ImageNet normalization, identical to the previous hardcoded behavior.

### Discussion points

1. **`normalize: Any` in schema.py.** OmegaConf's structured configs do not support unions like `dict | bool | None`. Using `Any` is the pragmatic choice here — it allows `null`, `false`, and `{mean, std}` to all pass through correctly, while validation is handled explicitly in `CropModel.normalize()`. Open to other approaches if preferred.

2. **Validation test for missing `mean` or `std`.** I added a test that verifies a `ValueError` is raised when a user provides `mean` without `std` (or vice versa). This felt worth catching early rather than letting it fail deeper in the transform pipeline with an unclear error. Let me know if this is considered unnecessary overhead.

## Related Issue(s)

Closes #1347

**How AI was used:**
- Understanding PR #1313's code changes and the threading pattern for `resize_interpolation`
- Breaking down the issue into sub-problems (identifying the two independent normalization sites, tracing call chains)
- Learning how OmegaConf handles structured configs and union types
- Code review and improving overall code quality after I proposed and implemented the change flow

I studied the issue and #1313 to understand the full context, proposed the threading approach and config interface myself, then used AI to review my implementation and catch edge cases (like the OmegaConf `DictConfig` vs `dict` issue and the missing `mean`/`std` validation).

### CODE REVIEW

File: src/deepforest/predict.py [hunk 1]
```diff
@@ -275,11 +276,19 @@ def _predict_crop_model_(
...
     if transform is None and hasattr(crop_model, "config"):
         resize = crop_model.config.get("cropmodel", {}).get("resize", [224, 224])
+        norm_cfg = crop_model.config.get("cropmodel", {}).get("normalize", None)
+        if norm_cfg is False:
+            normalize = False
+        elif norm_cfg is not None and "mean" in norm_cfg and "std" in norm_cfg:
+            normalize = transforms.Normalize(
+                mean=list(norm_cfg["mean"]), std=list(norm_cfg["std"])
+            )
         expand = crop_model.config.get("cropmodel", {}).get("expand", 0)
```
[the remark this answers is not in the document]
Comment (musaqlain) [reply src/deepforest/predict.py]: Good point. Replaced the duplicated parsing with a call to `crop_model.normalize()` so both paths share the same validation.

File: src/deepforest/predict.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import numpy as np
 import pandas as pd
 import torch
+from torchvision import transforms
```
[the remark this answers is not in the document]
Comment (musaqlain) [reply src/deepforest/predict.py]: fixed

**Review by bw4sz — APPROVED** [review]
This looks good to me, thank you. We don't yet have a backbone that is non-resnet, but I can imagine we will have some shortly.
